http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -1020,4 +1129,15 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
   def getWsmJobId(jobName: String, resourceId: WsmControlledResourceId): WsmJobId = WsmJobId(
     s"${jobName}-${resourceId.value.toString.take(10)}"
   )
+
+  private def buildWsmControlledResourceApiClient(implicit ev: Ask[F, AppContext]): F[ControlledAzureResourceApi] =
```

Comment:
I think there's a similar function in AKSInterpreter, is this something we could shift somewhere both classes could access?

Reply:
+1 to that, would it make sense to have this in the `WsmApiClientProvider` file?

Reply:
I think this makes sense here, since it lets the consumer of the client decide how to get the token (the AKSInterpreter gets the token with the `authProvider` instead of the `samDao`)